assignments/prototypes.js

```diff
@@ -102,9 +144,109 @@
   console.log(archer.greet()); // Lilith offers a greeting in Elvish.
   console.log(mage.takeDamage()); // Bruce took damage.
   console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+
 
   // Stretch task: 
   // * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.  
   // * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villain and one a hero and fight it out with methods!
\ No newline at end of file
+  // * Create two new objects, one a villain and one a hero and fight it out with methods!
+
+
+//Hero boi
+const Hero = function(attributes) {
+  Humanoid.call(this, attributes);
+}
+
+Hero.prototype = Object.create(Humanoid.prototype);
+Hero.prototype.attack = function(target) {
+  if(this.healthPoints > 0) {
+    if(target.healthPoints > 0) {
+      let damage = (Math.round(Math.random() * 10));
+      target.healthPoints = target.healthPoints - damage;
+      if (target.healthPoints > 0) {
+        console.log(`${this.name} attacked ${target.name} for ${damage} points! ${target.name} now has ${target.healthPoints} health`)
+      } else {
+        console.log(`${this.name} attacked ${target.name} for ${damage} points and destroyed ${target.name}!`)
```

Comment:
You could use the `destroy` and `takeDamage` functions you already have for these